lib/daru/index/index.rb

```diff
@@ -43,10 +43,6 @@ def each(&block)
       self
     end
 
-    def map(&block)
```

Comment:
The `Daru::Index` class is a super class for all other index classes (`MultiIndex`, `DateTimeIndex` and `CategoricalIndex`). The only way in which each element can be iterated over in these classes is by calling `to_a` on them since they utilize speciliazed data structures for storing their data. So `map` has been put here on purpose so that it can be shared by all the index classes that inherit from `Daru::Index`.